src/Util/HTTPSig.php

```diff
@@ -0,0 +1,400 @@
+<?php
+
+/**
+ * @file src/Util/HTTPSig.php
+ */
+namespace Friendica\Util;
+
+use Friendica\Core\Config;
+use Friendica\Database\DBM;
+use Friendica\Util\Crypto;
+use Friendica\Util\HTTPHeaders;
+use dba;
+
+/**
+ * @brief Implements HTTP Signatures per draft-cavage-http-signatures-07.
+ *
+ * Ported from Hubzilla: https://framagit.org/hubzilla/core/blob/master/Zotlabs/Web/HTTPSig.php
+ * 
+ * @see https://tools.ietf.org/html/draft-cavage-http-signatures-07
+ */
+
+class HTTPSig
+{
+	/**
+	 * @brief RFC5843
+	 *
+	 * @see https://tools.ietf.org/html/rfc5843
+	 *
+	 * @param string  $body The value to create the digest for
+	 * @param boolean $set  (optional, default true)
+	 *   If set send a Digest HTTP header
+	 * 
+	 * @return string The generated digest of $body
+	 */
+	public static function generateDigest($body, $set = true)
+	{
+		$digest = base64_encode(hash('sha256', $body, true));
+
+		if($set) {
+			header('Digest: SHA-256=' . $digest);
+		}
+		return $digest;
+	}
+
+	// See draft-cavage-http-signatures-08
+	public static function verify($data, $key = '')
+	{
+		$body      = $data;
+		$headers   = null;
+		$spoofable = false;
+		$result = [
+			'signer'         => '',
+			'header_signed'  => false,
+			'header_valid'   => false,
+			'content_signed' => false,
+			'content_valid'  => false
+		];
+
+		// Decide if $data arrived via controller submission or curl.
+		if (is_array($data) && $data['header']) {
+			if (!$data['success']) {
+				return $result;
+			}
+
+			$h = new HTTPHeaders($data['header']);
+			$headers = $h->fetcharr();
```

Comment:
Once you're done with the changes in the `HTTPHeaders` class, you can rename this to simply `fetch`.